src/NUnitFramework/framework/Constraints/ExactCountConstraintResult.cs

```diff
@@ -0,0 +1,62 @@
+using System.Collections;
+using NUnit.Framework.Internal;
+
+namespace NUnit.Framework.Constraints
+{
+    /// <summary>
+    /// Contain the result of matching a <see cref="ExactCountConstraint"/> against an actual value.
+    /// </summary>
+    internal sealed class ExactCountConstraintResult : ConstraintResult
+    {
+        /// <summary>
+        /// The list with all items of the <see cref="ExactCountConstraint"/>
+        /// </summary>
+        private readonly IEnumerable _elementList;
+
+        /// <summary>
+        /// The count of all items of the <see cref="ExactCountConstraint"/> (and of <see cref="_elementList"/>)
+        /// </summary>
+        private readonly int _itemCount;
+
+        /// <summary>
+        /// The count of matched items of the <see cref="ExactCountConstraint"/>
+        /// </summary>
+        private readonly int _matchCount;
+
+        /// <summary>
+        /// Constructs a <see cref="ConstraintResult"/> for a <see cref="ExactCountConstraint"/>.
+        /// </summary>
+        /// <param name="constraint">The Constraint to which this result applies.</param>
+        /// <param name="actualValue">The actual value to which the Constraint was applied.</param>
+        /// <param name="isSuccess">If true, applies a status of Success to the result, otherwise Failure.</param>
+        /// <param name="itemCount">Count of all items of the <see cref="ExactCountConstraint"/></param>
+        /// <param name="matchCount">Count of matched items of the <see cref="ExactCountConstraint"/></param>
+        internal ExactCountConstraintResult(IConstraint constraint, object actualValue, bool isSuccess, int itemCount, int matchCount)
+            : base(constraint, actualValue, isSuccess)
+        {
+            _elementList = ConstraintUtils.RequireActual<IEnumerable>(actualValue, nameof(actualValue));
+            _itemCount   = itemCount;
+            _matchCount  = matchCount;
+        }
+
+        /// <summary>
+        /// Write the actual value for a failing constraint test to a MessageWriter.
+        /// </summary>
+        /// <param name="writer">The writer on which the actual value is displayed</param>
+        public override void WriteActualValueTo(MessageWriter writer)
+        {
+            if (_elementList == null)
+            {
+                writer.Write("no items");
+                return;
+            }
+
+            writer.Write(_matchCount != 1 ? "{0} items <" : "{0} item <", _matchCount);
+
+            var writeCounter  = 1;
```

Comment:
I find the aligned `=` a little odd. The only problem I see is that when future contributors edit nearby, their editor will cause them grief by reformatting to remove the extra spaces and they'll have chose between painstakingly putting them back, or including changes over unrelated lines in the git blame history.